crates/flux-desugar/src/desugar.rs

```diff
@@ -1448,6 +1449,9 @@ trait DesugarCtxt<'genv, 'tcx: 'genv> {
             }
             surface::LitKind::Bool => Ok(fhir::Lit::Bool(lit.symbol == kw::True)),
             surface::LitKind::Str => Ok(fhir::Lit::Str(lit.symbol)),
+            surface::LitKind::Char => {
+                Ok(fhir::Lit::Char(u32::from(lit.symbol.as_str().parse::<char>().unwrap())))
```

Comment:
I'd try as much as possible to have them as `Char` so we don't lose the type invariant associated with it.

Reply:
I wonder if we can encode `Char` in fixpoint as string of size `1` ?

Reply:
🤔 that may be a good idea. I guess the best option depends on the theory functions for strings and how they relate to chars. For the time being, encoding char as integers is the easiest, it wouldn't be to hard to change it later.